src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonParameterInfo.cs

```diff
@@ -123,5 +123,9 @@ internal JsonParameterInfo(JsonParameterInfoValues parameterInfoValues, JsonProp
         internal JsonNumberHandling? NumberHandling => MatchingProperty.EffectiveNumberHandling;
         internal JsonTypeInfo JsonTypeInfo => MatchingProperty.JsonTypeInfo;
         internal bool ShouldDeserialize => !MatchingProperty.IsIgnored;
+        internal bool IsRequiredParameter =>
+            Options.RespectRequiredConstructorParameters &&
+            !HasDefaultValue &&
+            !IsMemberInitializer;
```

Comment:
Can you please explain why `!IsMemberInitializer` makes this evaluate as false?

From its remarks, what's an object initialize expression?
https://github.com/dotnet/runtime/blob/60edbf9371ff00349e9224fb0920c8bca51f1a4f/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonParameterInfo.cs#L81-L82

Reply:
It is members initialized via [object initializer expressions](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers#object-initializers). In the case of the source generator, `required` or `init` members can only be specified as object initializers so they are included as secondary parameters in the generated constructor delegate, roughly:
```C#
args => new MyPoco(args[0], args[1]) { RequiredProp = args[2], InitProp = args[3] };
```
The second type of parameters are identified by the `IsMemberInitializer` property.

For the case of this feature, we are only interested in non-optional parameters that are not initializers (because requiredness is handled differently in that case).